app/assets/javascripts/controllers/base_controller.js.coffee.erb

```diff
@@ -0,0 +1,39 @@
+Blog = angular.module("Blog")
+
+Blog.config ($routeProvider) ->
+  $routeProvider.otherwise(
+    templateUrl: '<%= asset_path('assets/welcome/index.html') %>',
+    controller: 'BaseController'
+)
+
+Blog.controller('BaseController', ($scope, User, Notification) ->
+  # Used a promise to avoid race conditions and the catch
+  # to clear the current user if something went wrong while
+  # attempting to get that user.
+  User.current().$promise.then (user) ->
+    $scope.currentUser = user
+  .catch () ->
+    delete $scope.currentUser
+
+
+
+  $scope.markedAsRead = false
+  $scope.markNotificationsAsRead = () ->
+    if !$scope.markedAsRead
+      $scope.markedAsRead = true
+      Notification.markAllRead()
+      $scope.notifications = [] 
+
+  $scope.getNotifications = () ->
+    $scope.notifications = Notification.query(query: {read: false})
```

Comment:
unless I'm wrong in assuming, Notifications are attached to user_id too, so you might need to pass user_id: $scope.current_user.id into this query too, or everyone will see all the unread notifications all the time